api/gateway.go

```diff
@@ -41,6 +45,10 @@ func (srv *Server) gatewayAddHandler(w http.ResponseWriter, req *http.Request, p
 // gatewayRemoveHandler handles the API call to remove a peer from the gateway.
 func (srv *Server) gatewayRemoveHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 	addr := modules.NetAddress(ps.ByName("netaddress"))
+	if err := addr.IsValid(); err != nil {
+		writeErorr(w, "invalid address: "+err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
Oops, you were right. I forgot that map access does not insert an object there with the zero value like it does in C++. Old habits.